### PR TITLE
Fix index filtering in follow info api.

### LINKED ISSUE #37738: [CCR] Follower info API doesn't return paused status
I have two active follower indices: `copy-of-kibana_sample_data_logs`, `copy-of-kibana_sample_data_flights`.

Then I send a request to pause the second one:
```
POST /copy-of-kibana_sample_data_flights/_ccr/pause_follow
```

Follower stats API does not show the index that was paused (correct behavior):
```
GET /_ccr/stats

{
  "auto_follow_stats" : {
    "number_of_failed_follow_indices" : 0,
    "number_of_failed_remote_cluster_state_requests" : 0,
    "number_of_successful_follow_indices" : 2,
    "recent_auto_follow_errors" : [ ],
    "auto_followed_clusters" : [
      {
        "cluster_name" : "prod2",
        "time_since_last_check_millis" : 35108,
        "last_seen_metadata_version" : 32
      }
    ]
  },
  "follow_stats" : {
    "indices" : [
      {
        "index" : "copy-of-kibana_sample_data_logs",
        "shards" : [
          {
            "remote_cluster" : "prod2",
            "leader_index" : "kibana_sample_data_logs",
            "follower_index" : "copy-of-kibana_sample_data_logs",
            "shard_id" : 0,
            "leader_global_checkpoint" : 14004,
            "leader_max_seq_no" : 14004,
            "follower_global_checkpoint" : 14004,
            "follower_max_seq_no" : 14004,
            "last_requested_seq_no" : 14004,
            "outstanding_read_requests" : 1,
            "outstanding_write_requests" : 0,
            "write_buffer_operation_count" : 0,
            "write_buffer_size_in_bytes" : 0,
            "follower_mapping_version" : 2,
            "follower_settings_version" : 1,
            "total_read_time_millis" : 6209,
            "total_read_remote_exec_time_millis" : 6086,
            "successful_read_requests" : 29,
            "failed_read_requests" : 0,
            "operations_read" : 14005,
            "bytes_read" : 13606642,
            "total_write_time_millis" : 3231,
            "successful_write_requests" : 29,
            "failed_write_requests" : 0,
            "operations_written" : 14005,
            "read_exceptions" : [ ],
            "time_since_last_read_millis" : 39676
          }
        ]
      }
    ]
  }
}
```

However, the follower info API still returns `status: "active"` for it:
```
GET /_all/_ccr/info

{
  "follower_indices" : [
    {
      "follower_index" : "copy-of-kibana_sample_data_logs",
      "remote_cluster" : "prod2",
      "leader_index" : "kibana_sample_data_logs",
      "status" : "active",
      "parameters" : {
        "max_read_request_operation_count" : 5120,
        "max_read_request_size" : "32mb",
        "max_outstanding_read_requests" : 12,
        "max_write_request_operation_count" : 5120,
        "max_write_request_size" : "9223372036854775807b",
        "max_outstanding_write_requests" : 9,
        "max_write_buffer_count" : 2147483647,
        "max_write_buffer_size" : "512mb",
        "max_retry_delay" : "500ms",
        "read_poll_timeout" : "1m"
      }
    },
    {
      "follower_index" : "copy-of-kibana_sample_data_flights",
      "remote_cluster" : "prod2",
      "leader_index" : "kibana_sample_data_flights",
      "status" : "active",
      "parameters" : {
        "max_read_request_operation_count" : 5120,
        "max_read_request_size" : "32mb",
        "max_outstanding_read_requests" : 12,
        "max_write_request_operation_count" : 5120,
        "max_write_request_size" : "9223372036854775807b",
        "max_outstanding_write_requests" : 9,
        "max_write_buffer_count" : 2147483647,
        "max_write_buffer_size" : "512mb",
        "max_retry_delay" : "500ms",
        "read_poll_timeout" : "1m"
      }
    }
  ]
}
```

cc @martijnvg

### PR DESCRIPTION
**Pull request by martijnvg**
The filtering by follower index was completely broken.
Also the wrong persistent tasks were selected, causing the
wrong status to be reported.

The fix itself is small, but this PR adds more tests.

Closes #37738

Note: marking this issue as non-issue (instead of a bug), because the follow info api has not been released yet.

### CODE REVIEW

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/1

**Review by jasontedor — APPROVED** [review]
LGTM.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java [hunk 1]
```diff
@@ -65,19 +65,28 @@ protected void masterOperation(FollowInfoAction.Request request,
...
+            IndexMetaData indexMetaData = state.metaData().index(index);
             Map<String, String> ccrCustomData = indexMetaData.getCustomData(Ccr.CCR_CUSTOM_METADATA_KEY);
             if (ccrCustomData != null) {
                 Optional<ShardFollowTask> result;
                 if (persistentTasks != null) {
                     result = persistentTasks.taskMap().values().stream()
                         .map(persistentTask -> (ShardFollowTask) persistentTask.getParams())
-                        .filter(shardFollowTask -> concreteFollowerIndices.isEmpty() ||
-                            concreteFollowerIndices.contains(shardFollowTask.getFollowShardId().getIndexName()))
+                        .filter(shardFollowTask -> index.equals(shardFollowTask.getFollowShardId().getIndexName()))
```
Comment (dnhatn) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java]: I think we need to filter ShardFollowTask task before casting. Maybe use `persistentTasks.findTasks` API.

I will make a PR to inject other persistent tasks randomly in our IT tests.

**Review by dnhatn — COMMENTED** [review]
Thanks @martijnvg. I left a comment.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java]: Good catch! I will use that method instead.

> I will make a PR to inject other persistent tasks randomly in our IT tests.

:+1:

**Comment by martijnvg** [comment]
@dnhatn I've updated the PR.

**Review by dnhatn — APPROVED** [review]
LGTM.

**Pull request merged**